docs/conditional_rendering.md

```diff
@@ -0,0 +1,268 @@
+        rx.text(f"Age: {CondComplexState.age}"),
+        rx.cond(
+            (CondComplexState.age >= 18) & (CondComplexState.age <=65),
+            rx.text("You can work!", color="green"),
+            rx.text("You cannot work!", color="red"),
+        ),
+    )
+
+```
+
+```python eval
+docdemo_from(CondComplexState, component=cond_complex_example, imports=["import random"])
+```
+
+
+
+
+
+## Reusing Cond
+
+
+We can also reuse a `cond` component several times by defining it within a function that returns a `cond`. 
+
+In this example we define the function `render_item`. This function takes in an `item`, uses the `cond` to check if the item `is_packed`. If it is packed it returns the `item_name` with a `✔` next to it, and if not then it just returns the `item_name`.
+
+```python exec
+class ToDoListItem(rx.Base):
+    item_name: str
+    is_packed: bool
+
+class CondRepeatState(rx.State):
+    to_do_list: list[ToDoListItem] = [
+        ToDoListItem(item_name="Space suit", is_packed=True), 
+        ToDoListItem(item_name="Helmet", is_packed=True),
+        ToDoListItem(item_name="Back Pack", is_packed=False),
+        ]
+
+
+def render_item(item: [str, bool]):
+    return rx.cond(
+        item.is_packed, 
+        rx.list_item(item.item_name + ' ✔'),
+        rx.list_item(item.item_name),
+        )
+
+def packing_list():
+    return rx.vstack(
+        rx.text("Sammy's Packing List", as_="strong"),
+        rx.list(rx.foreach(CondRepeatState.to_do_list, render_item)),
+    )
+
+```
+
+```python eval
+docdemo_from(ToDoListItem, CondRepeatState, render_item, component=packing_list)
+```
+
+
+## Nested Conditional 
+
+In python we can have an `if` statement that then has several `elif` statements before finishing with an `else`. This is also possible in reflex using nested `cond` components. Here we have three numbers and we are checking which of the three is the largest. If any two of them are equal then we return that `Some of the numbers are equal!`. 
+
+The reflex code that follows is logically identical to doing the following in python:
+
+```python
+a = 8
+b = 10
+c = 2
+
+if((a>b and a>c) and (a != b and a != c)): 
+	print(a, " is the largest!") 
+elif((b>a and b>c) and (b != a and b != c)): 
+	print(b, " is the largest!") 
+elif((c>a and c>b) and (c != a and c != b)): 
+	print(c, " is the largest!") 
+else: 
+	print("Some of the numbers are equal!") 
+```
+
+
+```python exec
+import random
+
+
+class CNS(rx.State):
+    # CNS: CondNestedState
+    a: int = 8
+    b: int = 10
+    c: int = 2
+    
+
+    def change(self):
+        self.a = random.randint(0, 10)
+        self.b = random.randint(0, 10)
+        self.c = random.randint(0, 10)
+
+
+def cond_nested_example():
+    return rx.vstack(
+        rx.button("Toggle", on_click=CNS.change),
+        rx.text(f"a: {CNS.a}, b: {CNS.b}, c: {CNS.c}"),
+        rx.cond(
+            ((CNS.a > CNS.b) & (CNS.a > CNS.c)) & ((CNS.a != CNS.b) & (CNS.a != CNS.c)),
```

Comment:
The combination of nesting and 4 conditionals check is making this example to hard to follow, a more simple example of nesting would make it clearer

Reply:
created a new example above it that is much simpler

Reply:
Nice looks good